### PR TITLE
fix: correct PME reciprocal virial shape for single-system batches

### PR DESCRIPTION
**Pull request by zubatyuk**
# ALCHEMI Toolkit-Ops Pull Request

## Description

Fix incorrect virial tensor shape returned by `particle_mesh_ewald` when `batch_idx` is provided with a single system (B=1). The reciprocal-space virial was `(nx, 3, 3)` instead of `(1, 3, 3)` because `generate_k_vectors_pme` strips the batch dimension via `squeeze(0)` when B=1, and the downstream batched einsum misinterprets the mesh grid dimension as the batch axis.

## Type of Change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Performance improvement
- [ ] Documentation update
- [ ] Refactoring (no functional changes)
- [ ] CI/CD or infrastructure change

## Related Issues

Fixes #49

## Changes Made

- Add `unsqueeze(0)` guards for `k_sq_acc` and `k_vecs_acc` in Torch `_compute_pme_reciprocal_virial` when `is_batch=True` and the batch dimension is missing.
- Add equivalent `jnp.expand_dims` guards in JAX `_compute_pme_reciprocal_virial`.
- Add regression tests for single-system batch virial shape in both Torch and JAX test suites.

## Testing

- [x] Unit tests pass locally (`make pytest`)
- [x] Linting passes (`make lint`)
- [x] New tests added for new functionality meets coverage expectations?

## Checklist

- [x] I have read and understand the [Contributing Guidelines](../CONTRIBUTING.md)
- [ ] I have updated the [CHANGELOG.md](../CHANGELOG.md)
- [x] I have performed a self-review of my code
- [x] I have added docstrings to new functions/classes
- [ ] I have updated the documentation (if applicable)

## Additional Notes

The fix mirrors an existing pattern in `_batch_pme_green_structure_factor`, which already guards against a missing batch dimension on `k_squared` with `unsqueeze(0)`. Multi-system batches (B>1) and non-batched calls (`batch_idx=None`) are unaffected.

### CODE REVIEW

File: nvalchemiops/jax/interactions/electrostatics/pme.py [hunk 1]
```diff
@@ -613,6 +613,11 @@ def _compute_pme_reciprocal_virial(
     k_sq_acc = k_squared.astype(acc_dtype)
     alpha_acc = alpha.astype(acc_dtype)
 
+    # generate_k_vectors_pme squeezes the batch dim when B=1; restore it so
```
Comment (WardLT) [inline nvalchemiops/jax/interactions/electrostatics/pme.py:616]: Did you also consider having `generate_k_vectors_pme` not squeeze the batch dimension away? I could see an argument for having the batch dimension always remain present, but don't have as firm of a grip on the use case for `generate_k_vectors_pme`

Comment (WardLT) [reply nvalchemiops/jax/interactions/electrostatics/pme.py:616]: I just looked over the `generate_k_vectors_pme` function and am actually going to argue for changing that function instead of this one. A "batched" input tensor gets squashed to a "single" tensor if the batch size happens to be 1 and changing the dimensionality without a user's consent seems undesirable.

Comment (zubatyuk) [reply nvalchemiops/jax/interactions/electrostatics/pme.py:616]: Changing `generate_k_vectors_pme` would change the behavior of a public API for batched inputs with B == 1. I agree that preserving the batch dimension is arguably a cleaner contract, but for this issue I’d prefer a focused fix in the PME consumer path, so we don’t introduce a broader API change in this bugfix.

Comment (dallasfoster) [reply nvalchemiops/jax/interactions/electrostatics/pme.py:616]: Agreed, we can reconsider the behavior in a future PR.

**Comment by zubatyuk** [comment]
linting ok

**Review by dallasfoster — APPROVED** [review]
